### ISSUE TITLE
Error when trying to use list prepend to replace pre-* commands in project.conf

### ISSUE THREAD

**Issue by cs-shadow** [cs-shadow/buildstream#131]
In GitLab by @hashpling on Oct 24, 2017, 08:03

Following issue #121, I was looking to replace a use of a `pre-` command in the `elements` section of a project.conf. The intention is to add an extra step project-wide for a particular kind of build, in this case the built in "autotools" kind.

Very cut down, this project.conf demonstrates the issue:

```yaml
name: testcase

# [...]

elements:
  autotools:
    config:
      pre-configure-commands:
      - echo "About to configure!"
```

This works correctly and my custom command is run before the configure step. If I change this as follows I get an error: "Error loading project: /src/project.conf [line 17 column 8]: Attempt to override non-existing list".

```yaml
elements:
  autotools:
    config:
      configure-commands:
        (<):
        - echo "About to configure!"
```

If I remove the `(<):` line then my custom command replaces the autotools plugin `configure-commands` and the project does not build correctly.

Is this a bug or have I misunderstood the configuration composition or the list directives?

**Comment by cs-shadow** [comment]
In GitLab by @hashpling on Oct 24, 2017, 08:13

changed the description

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Oct 24, 2017, 08:14

This is not a misunderstanding, it's a bug (and I believe you mean #121)

This [test case](https://gitlab.com/BuildStream/buildstream/commit/c20b15417f7519a0bfb481aeaa4fa5307a6fdaf2#fd5bd2182912630be9c46aec4ead948160e812cd_250_299) was intended to catch this kind of scenario, but I guess it slipped through.

Thanks for finding this !

**Comment by cs-shadow** [comment]
In GitLab by @hashpling on Oct 24, 2017, 08:14

Yes, that's what I meant, I've edited to refer to the issue, not the merge request of the same number.

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Oct 25, 2017, 11:06

mentioned in commit b14a0c575991f3b75aa15b21b2652186b62d0a09

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on Oct 25, 2017, 11:06

closed via commit b648bcab88340ea470098c9d2af60d3f556a51d6

**Comment by cs-shadow** [comment]
In GitLab by @tlater on Oct 25, 2017, 15:40

mentioned in commit b648bcab88340ea470098c9d2af60d3f556a51d6
